src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -41,26 +41,34 @@ public new void Free(List<T> list)
         private static readonly ListPool<INamespaceOrTypeSymbol> s_namespaceOrTypeListPool = new ListPool<INamespaceOrTypeSymbol>();
 
         /// <summary>
-        /// Creates an id string used by external documentation comment files to identify declarations
-        /// of types, namespaces, methods, properties, etc.
+        /// Creates an id string used by external documentation comment files to identify declarations of types,
+        /// namespaces, methods, properties, etc.
         /// </summary>
-        public static string CreateDeclarationId(ISymbol symbol)
+        /// <exception cref="ArgumentNullException">If <paramref name="symbol"/> is <see langword="null"/>.</exception>
+        /// <returns>The documentation comment Id for this symbol, if it can be created. <see langword="null"/> if it cannot be.</returns>
+        public static string? CreateDeclarationId(ISymbol symbol)
         {
             if (symbol == null)
             {
                 throw new ArgumentNullException(nameof(symbol));
             }
 
-            var builder = new StringBuilder();
-            var generator = new DeclarationGenerator(builder);
-            generator.Visit(symbol);
-            return builder.ToString();
+            var builder = PooledStringBuilder.GetInstance();
```

Comment:
>PooledStringBuilder.GetInstance() [](http://example.com/codeflow?start=26&length=33)

It looks like the switch to ```PooledStringBuilder``` is unrelated to the stated goal of the PR. We prefer to not bundle unrelated changes in a single PR. While, I don't see anything wrong in the change itself, I think it should be reverted and submitted as a separate PR.

Reply:
Removed.  Will add in separate pr.